### PR TITLE
dev-libs/atk-2.30.0, app-accessibility/at-spi2-{core,atk}-2.30.0 bumps

### PR DESCRIPTION
**Pull request by mattst88**
All ported to Meson.

### CODE REVIEW

File: app-accessibility/at-spi2-atk/at-spi2-atk-2.30.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+GNOME2_LA_PUNT="yes"
```
Comment (leio) [inline app-accessibility/at-spi2-atk/at-spi2-atk-2.30.0.ebuild]: Makes no sense with meson; rest of this ebuild is not reviewed

File: app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+LICENSE="LGPL-2+"
+SLOT="2"
+IUSE="X +introspection"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~x64-macos ~x86-macos"
+
+RDEPEND="
+	>=dev-libs/glib-2.36:2[${MULTILIB_USEDEP}]
+	>=sys-apps/dbus-1[${MULTILIB_USEDEP}]
+	introspection? ( >=dev-libs/gobject-introspection-1.32.0:= )
```
Comment (leio) [inline app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild]: meson builds need at least >=1.54 to not have potential failures in GIR generation

File: app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+GNOME2_LA_PUNT="yes"
```
Comment (leio) [inline app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild]: Makes no sense with meson

File: app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	>=sys-apps/dbus-1[${MULTILIB_USEDEP}]
+	introspection? ( >=dev-libs/gobject-introspection-1.32.0:= )
+	X? (
+		x11-libs/libX11[${MULTILIB_USEDEP}]
+		x11-libs/libXtst[${MULTILIB_USEDEP}]
+		x11-libs/libXi[${MULTILIB_USEDEP}]
+	)
+"
+DEPEND="${RDEPEND}
+	>=dev-util/gtk-doc-am-1.25
```
Comment (leio) [inline app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild]: gtk-doc-am is autotools specific

File: dev-libs/atk/atk-2.30.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+RDEPEND="
+	>=dev-libs/glib-2.34.3:2[${MULTILIB_USEDEP}]
+	introspection? ( >=dev-libs/gobject-introspection-1.32.0:= )
+"
+DEPEND="${RDEPEND}
+	>=dev-lang/perl-5
+	dev-util/glib-utils
+	>=dev-util/gtk-doc-am-1.25
+	>=virtual/pkgconfig-0-r1[${MULTILIB_USEDEP}]
+	nls? ( >=sys-devel/gettext-0.19.2 )
```
Comment (leio) [inline dev-libs/atk/atk-2.30.0.ebuild]: This is not optional in meson.build

File: dev-libs/atk/atk-2.30.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+IUSE="+introspection nls test"
+
+RDEPEND="
+	>=dev-libs/glib-2.34.3:2[${MULTILIB_USEDEP}]
+	introspection? ( >=dev-libs/gobject-introspection-1.32.0:= )
+"
+DEPEND="${RDEPEND}
+	>=dev-lang/perl-5
+	dev-util/glib-utils
+	>=dev-util/gtk-doc-am-1.25
```
Comment (leio) [inline dev-libs/atk/atk-2.30.0.ebuild]: gtk-doc-am is autotools only; This needs to be `IUSE=gtk-doc`, `gtk-doc? ( >=dev-util/gtk-doc-1.25)` and appropriate `$(meson_use_enable gtk-doc docs)` instead

File: dev-libs/atk/atk-2.30.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris ~x86-winnt"
+IUSE="+introspection nls test"
+
+RDEPEND="
+	>=dev-libs/glib-2.34.3:2[${MULTILIB_USEDEP}]
+	introspection? ( >=dev-libs/gobject-introspection-1.32.0:= )
+"
+DEPEND="${RDEPEND}
+	>=dev-lang/perl-5
```
Comment (leio) [inline dev-libs/atk/atk-2.30.0.ebuild]: I don't see any perl usage remaining with meson

File: app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+HOMEPAGE="https://wiki.gnome.org/Accessibility"
+
+LICENSE="LGPL-2+"
+SLOT="2"
+IUSE="X +introspection"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux ~x64-macos ~x86-macos"
+
+RDEPEND="
+	>=dev-libs/glib-2.36:2[${MULTILIB_USEDEP}]
+	>=sys-apps/dbus-1[${MULTILIB_USEDEP}]
```
Comment (leio) [inline app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild]: This dep was raised in meson.build; we want to be explicit with minimum deps, requiring at least what upstream build system requires

File: app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild (hunk 4 above)
Comment (leio) [reply app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild]: So probably IUSE=gtk-doc stuff here too.

File: dev-libs/atk/atk-2.30.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+	nls? ( >=sys-devel/gettext-0.19.2 )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-use-basename-instead-of-filename.patch
+)
+
+multilib_src_configure() {
+	local emesonargs=(
+		-Denable-introspection=$(multilib_native_usex introspection)
```
Comment (leio) [inline dev-libs/atk/atk-2.30.0.ebuild]: There is no such option in `meson_options.txt` and configure warns.

File: app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild (hunk 8 above)
Comment (mattst88) [reply app-accessibility/at-spi2-core/at-spi2-core-2.30.0.ebuild]: I'm going to bump the minimum required in the eclass, so that will resolve this issue.

**Comment by leio** [comment]
Merged with follow-up fix commits for each package

**Pull request closed without merging**